### PR TITLE
[associative.reqmts.general] Fix confusing local use of 'r'

### LINKED ISSUE #5134: [associative.reqmts.general] introduces `r` with two different meanings
[N4901](https://isocpp.org/files/papers/N4901.pdf) [associative.reqmts.general]/8.13:

> — `r` denotes a valid dereferenceable iterator to `a`,

But then /8.19 through /8.22 introduce `r` with a totally different meaning:

> — `kl` is a value such that `a` is partitioned (25.8) with respect to `c(r, kl)`, with `r` the key value of `e` and `e` in `a`;
> — `ku` is a value such that `a` is partitioned with respect to `!c(ku, r)`;
> — `ke` is a value such that `a` is partitioned with respect to `c(r, ke)` and `!c(ke, r)`, with `c(r, ke)` implying `!c(ke, r)`;
> — `kx` is a value such that
> — `a` is partitioned with respect to `c(r, rx)` and `!c(kx, r)`, with `c(r, kx)` implying `!c(kx, r)`, and
> — `kx` is not convertible to either `iterator` or `const_iterator`; and

This is confusing; a different name should be introduced.

### PR DESCRIPTION
**Pull request by jensmaurer**
Fixes #5134

### CODE REVIEW

File: source/containers.tex [hunk 1]
```diff
@@ -1710,21 +1710,24 @@
 \tcode{c} denotes a possibly \keyword{const} value of type \tcode{X::key_compare};
 \item
 \tcode{kl} is a value such that \tcode{a} is partitioned\iref{alg.sorting}
-with respect to \tcode{c(r, kl)}, with \tcode{r} the key value of \tcode{e}
```
Comment (tkoeppe) [inline source/containers.tex:1713]: I appreciate the heavy maths jargon here, and even though it resonates on a personal level, I think it's maybe a bit unnecessary in a general text. What do you think of changing this to the more normal "where `r` is the key value of `e` and `e` is in [or "an element of"] `a`"?

Comment (jensmaurer) [reply source/containers.tex:1713]: Well, the existing text already has the phrase; we're just replacing "r" with "x" and making sure the phrase is duplicated everywhere "x" is used.

The later bullets already contain a "with" phrase, so continuing with "where" is a bit inconsistent.

Comment (jensmaurer) [reply source/containers.tex:1713]: (If we want to improve clarity here, adding "all" would probably also be helpful.)

Comment (tkoeppe) [reply source/containers.tex:1713]: It's the "consistency with a bad idea" thing... OK, let's make the change as is, and maybe we can rephrase everything afterwards.

**Comment by jensmaurer** [comment]
@jwakely , @StephanTLavavej , please have a look.

**Review by jwakely — APPROVED** [review]
As I noted in #5134 it would be a smaller edit to change the first `r` which is the one that introduced the problem (the cases being changed were there first, since C++14).

But this is also OK.

**Review by StephanTLavavej — APPROVED** [review]

**Comment by jensmaurer** [comment]
@tkoeppe, this seems ready.

File: source/containers.tex (hunk 1 above)
Comment (tkoeppe) [reply source/containers.tex:1713]: (Note to selves: revisit this.)

**Pull request merged**